### ISSUE TITLE
NULL values in a character varying postgis columns are shown as if they were blank

### ISSUE THREAD

**Issue by qgib** [qgis/QGIS#25439]
Author Name: **Alessandro Pasotti** (@elpaso)
Original Redmine Issue: [17542](https://issues.qgis.org/issues/17542)
Affected QGIS version: master
Redmine category:attribute_table

---
It is not possible to tell the difference while editing.

---

- [qgis-blank-null.png](https://issues.qgis.org/attachments/download/11744/qgis-blank-null.png) (Alessandro Pasotti)

**Comment by Meibes** [comment]
This bug still exists and does also affect the text-type. It also leads to some very unexpected results if you try to select features by expression or compare different layers.

Here is a test setup:

CREATE TABLE with_null(
    id SMALLSERIAL,
    description text
);

CREATE TABLE without_null(
    id SMALLSERIAL,
    description text
);

INSERT INTO with_null (description) VALUES (NULL);
INSERT INTO with_null (description) VALUES ('example');
INSERT INTO with_null (description) VALUES (NULL);

INSERT INTO without_null (description) VALUES ('');
INSERT INTO without_null (description) VALUES ('example');
INSERT INTO without_null (description) VALUES ('');

if you load these tables in qgis both tables look exactly the same, which is obviously wrong, because if you try to select features by the following expression, you should expect the same results:

"description" IS NOT NULL

But you get different results... see pictures

![with_null](https://user-images.githubusercontent.com/26258451/68388520-20aebf00-0161-11ea-820c-3e15bd25c151.png)
![with_null_selected](https://user-images.githubusercontent.com/26258451/68388521-21475580-0161-11ea-9da9-a669552059df.png)
![without_null](https://user-images.githubusercontent.com/26258451/68388522-21475580-0161-11ea-80e5-12509fed3aaf.png)
![without_null_selected](https://user-images.githubusercontent.com/26258451/68388524-21475580-0161-11ea-8765-a43590d93a76.png)

NULL values for integer values are displayed correctly, that is why I assumed it would also display NULL-values for text-types

other expressions also tend to give unexpected results like:  "description" LIKE ''

**Comment by gioman** [comment]
> This bug still exists and does also affect the text-type. It also leads to some very unexpected results if you try to select features by expression or compare different layers.

I agree that this very confusing for users.

**Comment by Pedro-Murteira** [comment]
@elpaso Hello, I've tested this issue and from what I understand this might be fixed.
I used the example provided by @Meibes, and when I loaded the tables onto QGIS, I could see the NULL values on "with_null" table, consequently I used the expression: "description" IS NOT NULL, and got the expected results. Did you come across this issue more recently?

![test_25439](https://user-images.githubusercontent.com/92830786/150337308-cb70db8a-c8aa-4d66-8be1-5ae62c71b3de.png)
